metagpt/provider/own_api.py

```diff
@@ -0,0 +1,133 @@
+import copy
+import os
+import requests
+import json
+import sys
+import pdb
+import json
+import requests
+import hashlib
+import uuid
+import time
+import traceback
+import re
+from pathlib import Path
+import pandas as pd
+import datetime
+import uuid
+import qianfan
+from qianfan.resources.typing import JsonBody
+from metagpt.configs.llm_config import LLMConfig, LLMType
+from metagpt.const import USE_CONFIG_TIMEOUT
+from metagpt.logs import log_llm_stream
+from metagpt.provider.base_llm import BaseLLM
+from metagpt.provider.llm_provider_registry import register_provider
+from metagpt.utils.cost_manager import CostManager
+from metagpt.utils.token_counter import (
+    QIANFAN_ENDPOINT_TOKEN_COSTS,
+    QIANFAN_MODEL_TOKEN_COSTS,
+)
+
+@register_provider(LLMType.SHAHE)
+class OwnLLM(BaseLLM):
+    """
+    create your own api
+    """
+    def __init__(self, config: LLMConfig):
+        self.config = config
+        self.use_system_prompt = False  
+        self.__init_ownapi()
+        self.cost_manager = CostManager(token_costs=self.token_costs)
```

Comment:
refs to `https://docs.deepwisdom.ai/main/en/guide/contribute/contribute_guide.html#before-submission` to use pre-commit before commit.